### PR TITLE
[BEAM-8575] Refactor test_do_fn_with_windowing_in_finish_bundle to work with Dataflow runner

### PR DESCRIPTION
**Pull request by liumomo315**
The original test assumes there is always one bundle. The assumption is not true on the Dataflow runner. Limit input to one single element to enforce that.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by liumomo315** [comment]
R: @y1chi

Hi Yichi, this is a refactoring of https://github.com/apache/beam/pull/10145 to make this test run on Dataflow runner. PTAL, thanks!

File: sdks/python/apache_beam/transforms/ptransform_test.py [hunk 1]
```diff
@@ -347,14 +347,14 @@ def finish_bundle(self):
...
-              | 'Start' >> beam.Create([x for x in range(3)])
+              | 'Start' >> beam.Create([1])
               | beam.ParDo(MyDoFn())
               | WindowInto(windowfn)
               | 'create tuple' >> beam.Map(
                   lambda v, t=beam.DoFn.TimestampParam, w=beam.DoFn.WindowParam:
                   (v, t, w.start, w.end)))
-    expected_process = [('process'+ str(x), Timestamp(5), Timestamp(4),
-                         Timestamp(6)) for x in range(3)]
+    expected_process = [('process'+ str(1), Timestamp(5), Timestamp(4),
```
Comment (y1chi) [inline sdks/python/apache_beam/transforms/ptransform_test.py]:
```suggestion
    expected_process = [('process1', Timestamp(5), Timestamp(4),
```

**Comment by y1chi** [comment]
Run Python PreCommit

**Comment by y1chi** [comment]
Run Python Dataflow ValidatesRunner

**Comment by y1chi** [comment]
> R: @y1chi
> 
> Hi Yichi, this is a refactoring of #10145 to make this test run on Dataflow runner. PTAL, thanks

Seems the annotation is removed, do you need to add the validate runner annotation back?

**Comment by liumomo315** [comment]
Thanks for the quick review, Yichi!

>> Seems the annotation is removed, do you need to add the validate runner annotation back?
Not really. The goal is to add this test to internal Google3 Dataflow tests. Making it a validate runner test is not necessary. Thanks!

**Comment by liumomo315** [comment]
@y1chi The precommit error does not seem to relate to this PR, shall I add a committer to merge this PR? Thanks!

**Comment by y1chi** [comment]
@angoenka Ankur could you help merge the PR

**Comment by y1chi** [comment]
Run Python PreCommit

File: sdks/python/apache_beam/transforms/ptransform_test.py (hunk 1 above)
Comment (angoenka) [reply sdks/python/apache_beam/transforms/ptransform_test.py]: @liumomo315 
Shall I add this suggestion to the change?

Comment (y1chi) [reply sdks/python/apache_beam/transforms/ptransform_test.py]: it's not important I think. we can merge without.

Comment (liumomo315) [reply sdks/python/apache_beam/transforms/ptransform_test.py]: Sorry I missed this comment earlier. Committed the suggestion. Thanks!

**Comment by angoenka** [comment]
Run Python PreCommit

**Pull request merged**